File: Python/35-Search-Insert-Position.py
class Solution:
    def searchInsert(self, nums, target):
        for i in range(len(nums)):
            if target == nums[i]:
                return i
            elif nums[i] > target:
                return i
            if i == len(nums)-1:
                return i+1


s = Solution()
print(s.searchInsert([1, 3, 5, 6], 5))

File: JavaScript/35-Search-Insert-Position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (nums, target) {
    let l = 0
    let r = nums.length - 1

    while (l <= r) {
        let mid = Math.floor((r + l) / 2)

        if (nums[mid] === target) return mid
        else if (nums[mid] > target) r = mid - 1
        else l = mid + 1
    }

    return l
};